pkg/webhooks/validation/controller/controller.go

```diff
@@ -350,8 +350,11 @@ func (c *Controller) reconcileRequest(req *reconcileRequest) error {
 	defer func() { scope.Debugf("Reconcile(exit)") }()
 
 	failurePolicy := kubeApiAdmission.Ignore
-	if c.o.RemoteWebhookConfig {
+	// If the istiod endpoint does not exist, we need to set the failurePolicy to Fail when the WebhookConfiguration been first added.
+	// this is to handle when Pilot is run without WebhookConfiguration
+	if c.o.RemoteWebhookConfig || strings.HasPrefix(req.description, "update event") {
```

Comment:
This seems like a scary check. I doubt its clear that `description` is now basically a stable API

Reply:
The `description` will return you the event as below:
2020-07-13T11:50:42.370150Z	info	validationController	Reconcile(enter): **add event** (admissionregistration.k8s.io/v1beta1, Kind=ValidatingWebhookConfiguration) istiod-istio-system
2020-07-13T11:50:42.370236Z	info	validationController	Endpoint istiod is not ready: resource not found
2020-07-13T11:50:42.523228Z	info	Namespace controller started
2020-07-13T11:50:42.901133Z	info	validationController	Successfully updated validatingwebhookconfiguration istiod-istio-system (failurePolicy=Ignore,resourceVersion=7202161)
2020-07-13T11:50:42.901207Z	info	validationController	Reconcile(enter): **update event** (admissionregistration.k8s.io/v1beta1, Kind=ValidatingWebhookConfiguration) istiod-istio-system
2020-07-13T11:50:42.910438Z	info	validationController	Successfully updated validatingwebhookconfiguration istiod-istio-system (failurePolicy=Fail,resourceVersion=7202163)

So the idea here is after the ValidatingWebhookConfiguration is added( add event), we can safely update the failure policy to FAIL.  

Another option is to add a flag in profile that we can know it is running without istiod endpoint and skip the check.   not want to introduce new flag so this is why I go to with this option.

Reply:
Is it possible to leverage existing flag like pilot.enabled=true|false for this?

Reply:
I am super confused how this can possibly work and there are no tests to validate it so I cannot approve without understanding it myself.

Can you explain the steps that are taken for single cluster and central istiod?

Right now it looks like this change will impact non-central istiod as well.. I don't see how the webhook being updated means we should mark it as policy=Fail